tests/Filesystem/JoinPathsHelperTest.php

```diff
@@ -21,6 +21,7 @@ public static function unixDataProvider()
     {
         yield ['app/Http/Kernel.php', join_paths('app', 'Http', 'Kernel.php')];
         yield ['app/Http/Kernel.php', join_paths('app', '', 'Http', 'Kernel.php')];
+        yield ['app/Http/Kernel.php', join_paths('app', null, 'Http', 'Kernel.php')];
```

Comment:
I don't believe this is needed, as it not intended as valid.

Reply:
my argument is none of the helper tests have any test against `null` value which led to the parameter being typed hinted. But adding one now would make it harder to know if this usecase is intended or not far in the future.

Reply:
I suppose that's fair. I think removing this test is fine then.

I'm definitely in favor of adding more explicit type hints for the L11 upgrade, where I would fully expect to have to update my project for these sorts of reasons.

Reply:
Again, this test shouldn't be added.

Reply:
The test has been removed (along with the other one).

Reply:
Also, while even if this PR get accept, you should look into moving to string since null isn't intended.

Reply:
We're working a separate effort internally to use PHPStan (currently on level 1, trying to get to level 2), which is forcing us to make upgrades like this. It's a large legacy project with a small team, so progress is slow, but happening.